internal/strace/strace.go

```diff
@@ -84,6 +84,8 @@ type Result struct {
 	files    map[string]*FileInfo
 	sockets  map[string]*SocketInfo
 	commands map[string]*CommandInfo
+	// Map to track all seen write buffers so that we don't duplicate writing files to disk.
+	allWriteBufferId map[string]int
```

Comment:
Consider `map[string]bool` or even `map[string]struct{}` here instead.

The empty struct is preferred.